allennlp/predictors/text_classifier.py

```diff
@@ -22,4 +25,15 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
         Runs the underlying model, and adds the ``"label"`` to the output.
         """
         sentence = json_dict["sentence"]
+        if isinstance(self._dataset_reader, StanfordSentimentTreeBankDatasetReader):
+            tokenizer = WordTokenizer()
+            sentence = [str(t) for t in tokenizer.tokenize(sentence)]
         return self._dataset_reader.text_to_instance(sentence)
+
+    @overrides
+    def predictions_to_labeled_instances(self,
+                                         instance: Instance,
+                                         outputs: Dict[str, np.ndarray]) -> List[Instance]:
+        label = np.argmax(outputs['probs'])
+        instance.add_field('label', LabelField(int(label), skip_indexing=True))
```

Comment:
You should probably have a copy here.

Reply:
sorry can you clarify this

Reply:
You're modifying your input, instead of copying the instance and leaving the original unchanged.  In the other implementations, there's a `deepcopy` to make sure the original is unmodified.  It's probably a good idea to keep that here.